### PR TITLE
Add exercise 2

### PR DESCRIPTION
**Pull request by Ado77**

### CODE REVIEW

File: exercise2/ado77/common/treeHelpers.mjs [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+const _padNode = (arrToFill, start, length, char) => {
+    if (arrToFill.length < (start + length)) {
+        arrToFill[start + length] = '';
+    }
+    arrToFill.fill(char, start, start + length);
+}
+
+export const buildTreeMap = (map = [], currentNode, x, y, depth) => {
+
+    if (map[y] == undefined)
```
Comment (abhurke) [inline exercise2/ado77/common/treeHelpers.mjs]:
```suggestion
    if (map[y] == undefined) {
      map[y] = [];
    }
```

Comment (abhurke) [reply exercise2/ado77/common/treeHelpers.mjs]: Would you like to consistently use braces?

File: exercise2/ado77/common/treeHelpers.mjs [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+
+    if (currentNode.rightSide != undefined) {
+        _padNode(map[y], x + 1, padLength - 1, '_');
+        buildTreeMap(map, currentNode.rightSide, x + padLength, y + 1, depth);
+    }
+    return map;
+}
+
+export const showTreeMap = (map) => {
+    // console.table(map);
```
Comment (abhurke) [inline exercise2/ado77/common/treeHelpers.mjs]: I'm not a big fan of commenting code and committing it.

File: exercise2/ado77/common/sizeOf.mjs [hunk 3]
```diff
@@ -0,0 +1,62 @@
+'use strict'
+
+var ECMA_SIZES = {
+    STRING: 2,
+    BOOLEAN: 4,
+    NUMBER: 8
+};
+// var Buffer = require('buffer').Buffer
```
Comment (abhurke) [inline exercise2/ado77/common/sizeOf.mjs]: What are your thoughts about committing commenting code?

File: exercise2/ado77/common/treeHelpers.mjs [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+}
+
+const _padNode = (arrToFill, start, length, char) => {
+    if (arrToFill.length < (start + length)) {
+        arrToFill[start + length] = '';
+    }
+    arrToFill.fill(char, start, start + length);
+}
+
+export const buildTreeMap = (map = [], currentNode, x, y, depth) => {
```
Comment (abhurke) [inline exercise2/ado77/common/treeHelpers.mjs]: Out of curiosity, why are you building a TreeMap?

File: exercise2/ado77/common/treeHelpers.mjs [hunk 5]
```diff
@@ -0,0 +1,159 @@
...
+        console.log(line);
+    }
+    console.log(' ');
+}
+
+export const findTopNodes = (map) => {
+    let topNodes = [];
+    for (let y = 0; y < map.length; y++) {
+        for (let x = 0; x < map[y].length; x++) {
+            if (topNodes[x] == undefined && map[y][x] != undefined && '/\\'.indexOf(map[y][x]) === -1) {
```
Comment (abhurke) [inline exercise2/ado77/common/treeHelpers.mjs]: I'm curious to understand what is this condition doing

File: exercise2/ado77/common/treeHelpers.mjs [hunk 6]
```diff
@@ -0,0 +1,159 @@
...
+        findTreeWidthAndDepth(currentNode.rightSide, currentWidth + 1, currentDepth + 1, offset);
+    }
+
+    return {
+        width: (offset.max + -offset.min + 2),
+        depth: offset.depth + 1
+    };
+}
+
+const _padNode = (arrToFill, start, length, char) => {
```
Comment (abhurke) [inline exercise2/ado77/common/treeHelpers.mjs]: What's the convention to start a function with an underscore?

**Comment by abhurke** [comment]
I usually add a README for my design.

File: exercise2/ado77/common/sizeOf.mjs (hunk 3 above)
Comment (Ado77) [reply exercise2/ado77/common/sizeOf.mjs]: Lol, if I'm honest, I don't use that file... I was curious about the size of my objects/arrays...

File: exercise2/ado77/common/treeHelpers.mjs (hunk 6 above)
Comment (Ado77) [reply exercise2/ado77/common/treeHelpers.mjs]: It is the convention for "private" in JS... which has no real private anything...

File: exercise2/ado77/common/treeHelpers.mjs (hunk 4 above)
Comment (Ado77) [reply exercise2/ado77/common/treeHelpers.mjs]: Why not? 😄 

You don't like the name? It's what I use to represent the location of nodes...

**Pull request merged**